extensions/cornerstone/src/components/WindowLevelActionMenu/VolumePresets.tsx

```diff
@@ -0,0 +1,106 @@
+import { Icon } from '@ohif/ui';
+import { ButtonEnums } from '@ohif/ui';
+import React, { ReactElement, useState, useEffect, useCallback } from 'react';
+import { Button, InputFilterText } from '@ohif/ui';
+import { ViewportPreset, VolumePresetsProps } from '../../types/ViewportPresets';
+
+export function VolumePresets({
+  presets,
+  viewportId,
+  commandsManager,
+  onClose,
+}: VolumePresetsProps): ReactElement {
+  const [filteredPresets, setFilteredPresets] = useState(presets);
+  const [searchValue, setSearchValue] = useState('');
+  const [selectedPreset, setSelectedPreset] = useState<ViewportPreset | null>(null);
+
+  const handleSearchChange = useCallback((value: string) => {
+    setSearchValue(value);
+  }, []);
+
+  const formatLabel = (label: string, maxChars: number) => {
+    return label.length > maxChars ? `${label.slice(0, maxChars)}...` : label;
+  };
+
+  useEffect(() => {
+    if (searchValue) {
+      const filtered = presets.filter(preset =>
+        preset.name.toLowerCase().includes(searchValue.toLowerCase())
+      );
+      setFilteredPresets(filtered);
+    } else {
+      setFilteredPresets(presets);
+    }
+  }, [searchValue, presets]);
```

Comment:
I think this filtering can be inside the handleSearchChange callback, and we don't need a useeffect here